common/elasticsearch/client/os2/client.go

```diff
@@ -352,28 +334,24 @@ func (c *OS2) Scroll(ctx context.Context, index, body, scrollID string) (*client
 }
 
 func (c *OS2) Search(ctx context.Context, index, body string) (*client.Response, error) {
-	resp, err := c.client.Search(
-		c.client.Search.WithContext(ctx),
-		c.client.Search.WithIndex(index),
-		c.client.Search.WithBody(strings.NewReader(body)),
-	)
+
+	resp, err := c.client.Search(ctx, &osapi.SearchReq{
+		Indices: []string{index},
+		Body:    strings.NewReader(body),
+	})
 
 	if err != nil {
 		return nil, fmt.Errorf("OpenSearch Search: %w", err)
 	}
 
-	defer closeBody(resp)
-	if resp.IsError() {
-		return nil, types.InternalServiceError{
-			Message: fmt.Sprintf("OpenSearch Search Error: %v", c.parseError(resp)),
-		}
+	if resp.Inspect().Response == nil {
+		return nil, fmt.Errorf("OpenSearch search response nil")
 	}
-
 	var osResponse response
-	if err := c.decoder.Decode(resp.Body, &osResponse); err != nil {
+	bodyBytes, err := io.ReadAll(resp.Inspect().Response.Body)
+	if err := c.decoder.Decode(bytes.NewReader(bodyBytes), &osResponse); err != nil {
```

Comment:
took me a while to notice this one, but: err not checked.
it's a real shame that `errcheck` doesn't catch this :|  nor do a few other "must handle error" tools.

staticcheck seems to have built a better one somewhere along the way though:
```
❯ staticcheck -checks SA4006 ./common/elasticsearch/...
common/elasticsearch/client/os2/client.go:351:2: this value of err is never used (SA4006)
```

---

tbh I think I'll vote for doing this everywhere:
```go
if err := c.decoder.Decode(scrollResp.Inspect().Response.Body, &osResponse); err != nil && !errors.Is(err, io.EOF) {
```
seems worth it to make fewer checks to remember.